docs/src/pages/guides/minimizing-bundle-size/minimizing-bundle-size.md

```diff
@@ -33,30 +42,67 @@ import TextField from '@material-ui/core/TextField';
-Anything not listed there should be considered **private**, and subject to change without notice.
-For example, the `Tabs` component is a public module while `TabIndicator` is private.
+
+Be aware that we only support first and second leve imports. Anything below is considered
+private and can cause module duplication in your bundle.
+
+````js
+// OK
+import { Add as AddIcon } from '@material-ui/icons';
+import { Tabs } from '@material-ui/core';
+//                                 ^^^^ 1st or top-level
+
+// OK
+import AddIcon from '@material-ui/icons/Add';
+import Tabs from '@material-ui/core/Tabs';
+//                                  ^^^^ 2nd level
+
+// NOT OK
+import TabIndicator from '@material-ui/core/Tabs/TabIndicator';
+//                                               ^^^^^^^^^^^^ 3rd level
+
 
 ### Option 2
 
-Another option is to keep using the shortened import like the following, but still have the size of the bundle optimized thanks to a **Babel plugin**:
+**Important note**:
+This is only supported for `@material-ui/icons`. We recommend this approach if
+you often restart your development build.
 
-```js
-import { Button, TextField } from '@material-ui/core';
-```
+Another option is to keep using named imports, but still have shorter
+start up times by using `babel` plugins.
 
 Pick one of the following plugins:
 
-- [babel-plugin-import](https://github.com/ant-design/babel-plugin-import) is quite customizable and with enough tweaks works with Material-UI.
-- [babel-transform-imports](https://bitbucket.org/amctheatres/babel-transform-imports) has a different api than a `babel-plugin-import` but does same thing.
-- [babel-plugin-lodash](https://github.com/lodash/babel-plugin-lodash) aims to work out of the box with all the `package.json`.
-
-**Important note**: Both of these options *should be temporary* until you add tree shaking capabilities to your project.
+- [babel-plugin-import](https://github.com/ant-design/babel-plugin-import) with the following configuration:
+  ```js
+  [
+    'babel-plugin-import',
+    {
+      libraryName: '@material-ui/icons',
+      libraryDirectory: 'esm', // or '' if your bundler does not support ES modules
+      camel2DashComponentName: false,
+    },
+  ];
+  ```
+- [babel-plugin-transform-imports](https://www.npmjs.com/package/babel-plugin-transform-import) has a different api than `babel-plugin-import` but does same thing.
+  ```js
+  [
+    'transform-imports',
+    {
+      '@material-ui/icons': {
+        transform: '@material-ui/icons/esm/${member}',
+        // for bundlers not supporting ES modules use:
+        // transform: '@material-ui/icons/${member}',
+      },
+    },
+  ];
+  ```
 
 ## ECMAScript
 
 The package published on npm is **transpiled**, with [Babel](https://github.com/babel/babel), to take into account the [supported platforms](/getting-started/supported-platforms/).
 
-We also publish a second version of the components to target **evergreen browsers**.
```

Comment:
Might be misunderstood that the `/es` build targets evergreen browsers. We currently ship syntax in that build that is not supported in all evergreen browsers.

Reply:
👍